### PR TITLE
Metrics around webhook leadership and hub startup/shutdown

### PR DESCRIPTION
**Pull request by lkemmerer**

### CODE REVIEW

File: src/main/java/com/flightstats/hub/app/ShutdownManager.java [hunk 1]
```diff
@@ -114,6 +117,11 @@ private void waitForLock() throws Exception {
         }
     }
 
+    private void reportEvent(String eventName) {
+        String[] tags = {"restart", "shutdown"};
+        statsdReporter.incrementCounter(eventName, tags);
+    }
```
Comment (manjula-chikkanagappa) [inline src/main/java/com/flightstats/hub/app/ShutdownManager.java]: This functionality in this method is used in various places and in different forms. would it possible to pull these things in to one place?

Comment (lkemmerer) [reply src/main/java/com/flightstats/hub/app/ShutdownManager.java]: The main reason I extracted it in a couple places was that I didn't want to duplicate the tags in each place I needed to report an event for a particular class. Hiding the tags from the calling method and keeping the tags close to the reporter (vs. a private static final string array) seemed to increase readability. I believe the tags are different in each of the classes that use the pattern, so I didn't see a way to DRY up the behavior any further. What did you have in mind?

Comment (manjula-chikkanagappa) [reply src/main/java/com/flightstats/hub/app/ShutdownManager.java]: An EventReporter which accepts  event name/type and tags. Here the tags will be encapsulated in the object where it belongs.

Comment (lkemmerer) [reply src/main/java/com/flightstats/hub/app/ShutdownManager.java]: Oh! So something like:
```
private final Map<EnumEventName, String[]> eventsToTags = new HashMap(...);
public void reportStartEvent(EnumEventName eventType) {
    statsdReporter.incrementCounter(eventType + ".start", eventsToTag.get(eventType));
}
```
?

**Pull request merged**